KEEP/etc/butch-optflags.sh

```diff
@@ -4,14 +4,38 @@
 # this file is sourced by the default butch build template
 
 isgcc3() {
-	local mycc="$CC"
-	[ -z "$mycc" ] && mycc=gcc
-	$mycc --version | grep "3.4.6" >/dev/null
+  local mycc="$CC"
+  [ "$STAGE" = "0" ] && \
+    [ -x "$butch_root_dir/bin/musl-gcc" ] && \
+    mycc="$butch_root_dir/bin/musl-gcc"
+  [ -z "$mycc" ] && mycc=gcc
+  $mycc --version | grep "3.4.6" >/dev/null
 }
 
-optcflags="-fdata-sections -ffunction-sections -Os -g0 -fno-unwind-tables -fno-asynchronous-unwind-tables -Wa,--noexecstack"
+optcflags="-mtune=generic -fdata-sections -ffunction-sections -Os -g0 -fno-unwind-tables -fno-asynchronous-unwind-tables -Wa,--noexecstack -pipe"
 optldflags="-s -Wl,--gc-sections -Wl,-z,relro,-z,now"
 
+if ! isgcc3; then
+  if [ "$HARDENED" = "1" ]; then
+    optcflags="$optcflags -fstack-protector-all -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4"
+    [ "$A" = "i486" ] && optldflags="-lssp_nonshared $optldflags"
+    [ "$A" = "i386" ] && optldflags="-lssp_nonshared $optldflags"
+  elif [ "$HARDENED" = 2 ] ; then #incompatible with static builds
+    optcflags="$optcflags -fstack-protector-all -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -pie -fPIE"
+    optldflags="$optldflags -pie"
+    [ "$A" = "i486" ] && optldflags="-lssp_nonshared $optldflags"
+    [ "$A" = "i386" ] && optldflags="-lssp_nonshared $optldflags"
+  fi
+
+  if [ "$BRUTE" = 2 ] ; then #more agressive optimizations
+    optcflags="$(echo $optcflags |sed 's,-O[a-z0-9] ,-O3 -flto -fPIC ,g')"
```

Comment:
why would you apply a sed to filter out flags from user-set CFLAGS ? also -fPIC might not be what the user wants